tests/test_tools: fails on `iotlab-m3` due to `core`/`cpu` regression

<!--
====================================
IF YOUR ISSUE IS RELATED TO SECURITY
====================================
please submit it to the security mailing-list security@riot-os.org.

If your issue is a question related to the usage of RIOT, please submit it to
the user mailing-list users@riot-os.org or to the developer mailing-list
devel@riot-os.org.
-->

#### Description
The test `tests/test_tools` fails on `iotlab-m3`. According to `git bisect` this is a regression introduced in https://github.com/RIOT-OS/RIOT/commit/0ff9e554eb712fe4f37ea46e232eaed9c20b9071.
<!--
Example: Cannot build gnrc_networking application for samr21-xpro board.
-->

#### Steps to reproduce the issue
```sh
iotlab-experiment submit -d 10 -n RIOT-test_tools -l 1,archi=m3:at86rf231+site=grenoble
IOTLAB_NODE=auto-ssh BOARD=iotlab-m3 make -C tests/test_tools/ flash test
```
<!--
Try to describe as precisely as possible here the steps required to reproduce
the issue. Here you can also describe your hardware configuration, the network
setup, etc.
-->

#### Expected results
Test succeeds (presented result from 0ff9e554eb712fe4f37ea46e232eaed9c20b9071~1
```
shellping
ssh -t lenders@grenoble.iot-lab.info 'socat - tcp:m3-101.grenoble.iot-lab.info:20000'
shellpong
true this should not be echoed
shellping
shellpong
toupper lowercase
LOWERCASE
getchar

getchar 0x0a
```
<!--
Example: The gnrc_networking application builds on samr21-xpro.
-->

#### Actual results
Test fails on `toupper` test case
```
shellping
ssh -t lenders@grenoble.iot-lab.info 'socat - tcp:m3-101.grenoble.iot-lab.info:20000'
shellping
shellpong
shellpong
true this should not be echoed
shellping
toupper lowercase
shellpong

Traceback (most recent call last):
  File "/home/mlenders/Repositories/RIOT-OS/RIOT-release/tests/test_tools/tests/01-run.py", line 83, in <module>
    sys.exit(run(testfunc))
  File "/home/mlenders/Repositories/RIOT-OS/RIOT-release/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/home/mlenders/Repositories/RIOT-OS/RIOT-release/tests/test_tools/tests/01-run.py", line 76, in testfunc
    _test_clean_output(child)
  File "/home/mlenders/Repositories/RIOT-OS/RIOT-release/tests/test_tools/tests/01-run.py", line 55, in _test_clean_output
    assert retline.strip() == 'LOWERCASE'
AssertionError
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT-release/tests/test_tools/../../Makefile.include:768: test] Error 1
```
<!--
Please paste or specifically describe the actual output.
-->

#### Versions
Every version after https://github.com/RIOT-OS/RIOT/commit/0ff9e554eb712fe4f37ea46e232eaed9c20b9071 until current master

```
Operating System Environment
----------------------------
         Operating System: "Arch Linux" 
                   Kernel: Linux 5.7.7-arch1-1 x86_64 unknown
             System shell: GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
             make's shell: GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)

Installed compiler toolchains
-----------------------------
               native gcc: gcc (GCC) 10.1.0
        arm-none-eabi-gcc: arm-none-eabi-gcc (Arch Repository) 10.1.0
                  avr-gcc: missing
         mips-mti-elf-gcc: missing
               msp430-gcc: missing
     riscv-none-embed-gcc: missing
     xtensa-esp32-elf-gcc: missing
   xtensa-esp8266-elf-gcc: missing
                    clang: clang version 10.0.0 

Installed compiler libs
-----------------------
     arm-none-eabi-newlib: "3.3.0"
      mips-mti-elf-newlib: missing
  riscv-none-embed-newlib: missing
  xtensa-esp32-elf-newlib: missing
xtensa-esp8266-elf-newlib: missing
                 avr-libc: missing (missing)

Installed development tools
---------------------------
                   ccache: missing
                    cmake: cmake version 3.17.3
                 cppcheck: missing
                  doxygen: 1.8.18
                      git: git version 2.27.0
                     make: GNU Make 4.3
                  openocd: Open On-Chip Debugger 0.10.0+dev-01089-g3bfe49266 (2020-02-26-14:18)
                   python: Python 3.8.3
                  python2: Python 2.7.18
                  python3: Python 3.8.3
                   flake8: 3.8.3 (mccabe: 0.6.1, pycodestyle: 2.6.0, pyflakes: 2.2.0) CPython 3.8.3 on
               coccinelle: missing
```
<!--
Operating system: Mac OSX, Linux, Vagrant VM
Build environment: GCC, CLang versions (you can run the following command from
the RIOT base directory: make print-versions).
-->

<!-- Thanks for contributing! -->

--- a/cpu/cortexm_common/thread_arch.c
+++ b/cpu/cortexm_common/thread_arch.c
@@ -462,6 +462,7 @@ void sched_arch_idle(void)
      * According to [this](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dai0321a/BIHJICIE.html),
      * dynamically changing the priority is not supported on CortexM0(+).
      */
+    unsigned state = irq_disable();
     NVIC_SetPriority(PendSV_IRQn, CPU_CORTEXM_PENDSV_IRQ_PRIO + 1);
     __DSB();
     __ISB();
@@ -471,5 +472,6 @@ void sched_arch_idle(void)
 #else
     __WFI();
 #endif
+    irq_restore(state);
     NVIC_SetPriority(PendSV_IRQn, CPU_CORTEXM_PENDSV_IRQ_PRIO);
 }

